spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/ErrorChannelTests.java

```diff
@@ -0,0 +1,140 @@
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.integration.annotation.InboundChannelAdapter;
+import org.springframework.integration.annotation.Poller;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.messaging.support.ErrorMessage;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.Assert;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration({ErrorChannelTests.TestSource.class, ErrorChannelTests.TestErrorSink1.class,
+		ErrorChannelTests.TestErrorSink2.class})
+public class ErrorChannelTests {
+
+	@Rule
+	public RedisTestSupport redisTestSupport = new RedisTestSupport();
+
+	@Autowired
+	@Bindings(TestErrorSink1.class)
+	private ErrorSink1 testErrorSink1;
+
+	@Autowired
+	@Bindings(TestErrorSink2.class)
+	private ErrorSink2 testErrorSink2;
+
+	@Test
+	public void testErrorChannelBinding() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(1);
+		MessageHandler errorMessageHandler = new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				Assert.isTrue(message instanceof ErrorMessage, "Message should be an instance of ErrorMessage");
+				Assert.isTrue(message.getPayload() instanceof MessagingException, "Message payload should be an instance" +
+						"of MessagingException");
+				Assert.isTrue(message.getPayload().toString()
+						.equals("org.springframework.messaging.MessagingException: test"));
+				latch.countDown();
+			}
+		};
+		testErrorSink1.errorInput().subscribe(errorMessageHandler);
+		testErrorSink2.errorInput().subscribe(errorMessageHandler);
+		assertTrue(latch.await(10, TimeUnit.SECONDS));
+	}
+
+	@EnableBinding(Source.class)
+	@EnableAutoConfiguration
+	@Import(RedisMessageChannelBinderConfiguration.class)
+	@PropertySource("classpath:/org/springframework/cloud/stream/config/errorchannel/source-channel.properties")
+	public static class TestSource {
+
+		@Bean
+		@InboundChannelAdapter(value = Source.OUTPUT, poller = @Poller(fixedDelay = "5000", maxMessagesPerPoll = "1"))
+		public MessageSource<String> timerMessageSource() {
+			return new MessageSource<String>() {
+				@Override
+				public Message<String> receive() {
+					throw new MessagingException("test");
+				}
+			};
+		}
+	}
+
+	@EnableBinding(ErrorSink1.class)
+	@EnableAutoConfiguration
+	@Import(RedisMessageChannelBinderConfiguration.class)
+	@PropertySource("classpath:/org/springframework/cloud/stream/config/errorchannel/errorsink1-channel.properties")
+	public static class TestErrorSink1 {
+
+	}
+
+	public interface ErrorSink1 {
```

Comment:
Now that I recall, there is one caveat here. Since we wanted to test this for multiple destinations (error channel being pub/sub), we have to create one more interface with the `@Input` annotation but with a different name than **input**. I think that was the original reason behind adding multiple explicit interfaces here. May be I will keep the default `Sink` and create just one more?